### PR TITLE
Fix issue when retrieving catalogs using filter

### PR DESCRIPTION
**Pull request by mikeletux**
No issues associated with this PR

# Description 
Small bug fix regarding `getCatalogByFilter` function.

# Detailed description
Since the function `getCatalogByFilter` now retrieves `adminCatalog`, the previous implementation was trying to retrieve an `adminCatalog` from a `catalog` HREF, and that was causing the issue.

Signed-off-by: Miguel Sama <msama@vmware.com>

### CODE REVIEW

**Review by vbauzysvmware — APPROVED** [review]
LGTM

**Review by adambarreiro — APPROVED** [review]

File: .changes/v3.6.0/817-bug-fixes.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Fix bug regarding `getCatalogByFilter` function [GH-817]
```
Comment (lvirbalas) [inline .changes/v3.6.0/817-bug-fixes.md:1]: Please put a more complete issue description as you have in PR summary.

**Review by lvirbalas — APPROVED** [review]
LGTM, but please elaborate the change log.

File: .changes/v3.6.0/817-bug-fixes.md (hunk 1 above)
Comment (vbauzysvmware) [reply .changes/v3.6.0/817-bug-fixes.md:1]: If I understand correctly, I would not add a changelog message at all, as this issue was introduced by PR which is in this release.

Comment (lvirbalas) [reply .changes/v3.6.0/817-bug-fixes.md:1]: We add changelog entries from govcd perspective - not Terraform, as govcd is used by other users too.

But, if this issue was introduced in this release, then this shouldn't need an entry at all, as @vbauzysvmware indicated.

**Review by Didainius — APPROVED** [review]

**Pull request merged**